src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilder.java

```diff
@@ -256,6 +256,12 @@ public Pod build() {
         if (jnlp.getResources() == null) {
             jnlp.setResources(new ContainerBuilder().editOrNewResources().addToRequests("cpu", new Quantity("100m")).addToRequests("memory", new Quantity("256Mi")).endResources().build().getResources());
         }
+        
+        // If the volume mounts of any container has been set to null, set it to empty list.
+        // Without this being done the code below would throw a NPE if such null existed.
+        pod.getSpec().getContainers().stream()
+               .filter(c -> c.getVolumeMounts() == null)
+               .forEach(c -> c.setVolumeMounts(new ArrayList<>()));
 
         // default workspace volume, add an empty volume to share the workspace across the pod
         if (pod.getSpec().getVolumes().stream().noneMatch(v -> WORKSPACE_VOLUME_NAME.equals(v.getName()))) {
```

Comment:
Or we could just check `v != null` here, right?

Reply:
At line 273 `c.getVolumeMounts().stream()` will then still throw a NPE. If you check that for null as well, then it will prevent the NPE, but not add the default volume mount as intended.